tests/spread/integration/gcc-14-aarch64-linux-gnu/test_hello.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+# spellchecker: ignore rootfs binutils
+
+arch=$(uname -m)
+cross=false
+if [[ "$arch" == "x86_64" ]]; then
+    cross=true
+elif [[ "$arch" == "aarch64" ]]; then
+    cross=false
+else
+    echo "Unsupported architecture: $arch"
+    exit 1
+fi
+
+if $cross; then
+    # TODO: We do not have libgcc-14-dev-amd64-cross for cross compilation yet
+    rootfs=$(mktemp -d)
+else
+    rootfs="$(install-slices gcc-14-aarch64-linux-gnu_minimal)"
+    ln -s aarch64-linux-gnu-as "$rootfs/usr/bin/as"
+    ln -s aarch64-linux-gnu-ld "$rootfs/usr/bin/ld"
+    ln -s aarch64-linux-gnu-gcc-14 "$rootfs/usr/bin/gcc"
+fi
+
+cat > "${rootfs}/hello.c" << EOF
+#include <stdio.h>
+int main() {
+    printf("Hello from C!\n");
+    return 0;
+}
+EOF
+
+if $cross; then
```

Comment:
This feels like two tests to me: A non-cross-compiling test and an incomplete cross-compiling test. I'm not sure the best way to split this, but I think I'd prefer to have at most a single conditional.

Reply:
yeah. agreed. this is a change i would look into only after we have the cross tests working though, since we then know what actually is different between the tests, and, if there is more than one conditional, split them. for now i'd leave it in one file to make it easier to read + make it obvious the `$cross` test is incomplete